src/format/opamTypes.mli

```diff
@@ -262,9 +273,9 @@ type 'a request = {
 
 (** user request action *)
 type user_action =
-  | Install of name_set (** The 'root' packages to be installed *)
-  | Upgrade of package_set (** The subset of packages to upgrade *)
-  | Reinstall of package_set
+  | Install of (name_set * switch_set) (** The 'root' packages to be installed *)
+  | Upgrade of (package_set * switch_set) (** The subset of packages to upgrade on each switch *)
+  | Reinstall of (package_set * switch_set)
```

Comment:
These feel like they should be `(name * switch)` sets. What are the reasons to have this limitation ?
(I understand it may be more awkward and require more changes to not have a package/name set directly available?)